### ISSUE TITLE
watchtower interferes with itself when upgrading itself

### ISSUE THREAD

**Issue by kevinlul** [containrrr/watchtower#747]
**Describe the bug**
I have a Compose deployment with each container labelled with a watchtower scope, so the watchtower instance started is only responsible for the containers in this stack. I read in the documentation that watchtower is able to restart itself, however, today at 16:26 Eastern (2021-01-06 21:26 UTC), a new `latest` was pushed to Docker Hub, and the following sequence of events occurred:

<details>
<summary>Cleaned logs</summary>

```
2021-01-06T21:26:43Z [I] Found new containrrr/watchtower:latest image
                         (sha256:2ddeb26d6d30132b59dd61939972e199b021b2a4b16a08a8e7efba947cd84fc1)
2021-01-06T21:26:46Z [I] Creating /emcee-tournament-bot_watchtower_1
2021-01-06T21:26:58Z [I] Removing image sha256:5e033eff56ed81c489cbcc59d548c147dd6d08259fe3bfa0993347bef6f1f768
2021-01-06T21:26:58Z [E] Error response from daemon: conflict: unable to delete 5e033eff56ed (cannot be forced)
                         - image is being used by running container eac5362f02b9
2021-01-06T21:26:59Z [I] Found multiple running watchtower instances. Cleaning up.
2021-01-06T21:26:59Z [I] Stopping /dKVlIoptpzKdNshheqXDgXROCDCKIYuL
                         (eac5362f02b921e9f5e8546f7b152a4b926f5efcf77265560ecacca8cca9e90e) with SIGTERM

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa924ff]

goroutine 14 [running]:
github.com/containrrr/shoutrrr/pkg/services/discord.doSend(0xc000035810, 0x49, 0x50, 0xc0003ec080, 0x7b, 0x7c, 0x0)
        /go/pkg/mod/github.com/containrrr/shoutrrr@v0.0.0-20200601144753-78bb9685bc2f/pkg/services/discord/discord.go:62 +0xbf
github.com/containrrr/shoutrrr/pkg/services/discord.(*Service).Send(0xc000314f60, 0xc000385340, 0x3a, 0xc00000ef78, 0x0, 0x0)
        /go/pkg/mod/github.com/containrrr/shoutrrr@v0.0.0-20200601144753-78bb9685bc2f/pkg/services/discord/discord.go:37 +0x2c7
github.com/containrrr/shoutrrr/pkg/router.sendToService.func1(0xc00038a900, 0xd2a700, 0xc000314f60, 0xc000385340, 0x3a, 0xc00000ef78)
        /go/pkg/mod/github.com/containrrr/shoutrrr@v0.0.0-20200601144753-78bb9685bc2f/pkg/router/router.go:78 +0x4f
created by github.com/containrrr/shoutrrr/pkg/router.sendToService
        /go/pkg/mod/github.com/containrrr/shoutrrr@v0.0.0-20200601144753-78bb9685bc2f/pkg/router/router.go:78 +0xda

2021-01-06T21:27:00Z [I] Waiting for running update to be finished...
2021-01-06T21:27:13Z [I] Found new containrrr/watchtower:latest image
                         (sha256:2ddeb26d6d30132b59dd61939972e199b021b2a4b16a08a8e7efba947cd84fc1)
2021-01-06T21:27:18Z [I] Creating /dKVlIoptpzKdNshheqXDgXROCDCKIYuL
2021-01-06T21:27:22Z [I] Found multiple running watchtower instances. Cleaning up.
2021-01-06T21:27:22Z [I] Stopping /BAXNVEVxexWISWZpPaPEVfaABwfYuemN
                         (eac5362f02b921e9f5e8546f7b152a4b926f5efcf77265560ecacca8cca9e90e) with SIGTERM
2021-01-06T21:27:36Z [I] Removing image sha256:5e033eff56ed81c489cbcc59d548c147dd6d08259fe3bfa0993347bef6f1f768
2021-01-06T21:27:36Z [E] Error response from daemon: conflict: unable to delete 5e033eff56ed (cannot be forced)
                         - image is being used by running container eac5362f02b9
2021-01-06T21:27:40Z [I] Removing image sha256:5e033eff56ed81c489cbcc59d548c147dd6d08259fe3bfa0993347bef6f1f768
2021-01-06T21:27:41Z [I] Starting Watchtower and scheduling first run: 2021-01-06 21:28:11 +0000 UTC m=+51.277998541
2021-01-06T21:27:58Z [I] Waiting for running update to be finished...
```

</details>

**To Reproduce**

Simplified version of the [actual compose file](https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/docker-compose.yaml) used:

```yaml
version: "3.8"
services:
  postgres:
    image: postgres:alpine
    labels:
      com.centurylinklabs.watchtower.scope: alphakretin/emcee-tournament-bot
    restart: unless-stopped
  bot:
    image: node:14-buster
    labels:
      com.centurylinklabs.watchtower.scope: alphakretin/emcee-tournament-bot
    restart: unless-stopped
  watchtower:
    image: containrrr/watchtower
    environment:
      WATCHTOWER_CLEANUP: 1
      WATCHTOWER_SCOPE: alphakretin/emcee-tournament-bot
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    labels:
      com.centurylinklabs.watchtower.scope: alphakretin/emcee-tournament-bot
    restart: unless-stopped
```

**Expected behavior**
I expected watchtower to update itself as it does with other containers, without sending a bunch of errors, fighting itself, and segfaulting. I also expected the updated container to have the same name as initialized by Compose (`emcee-tournament-bot_watchtower_1`), but this was not the case (actual: `dKVlIoptpzKdNshheqXDgXROCDCKIYuL`)

**Screenshots**
![image](https://user-images.githubusercontent.com/6320810/103832347-167d9e80-504c-11eb-99fc-683200b3a90b.png)

**Environment**
 - Platform: Ubuntu 18.04.2 LTS
 - Architecture: x86_64
 - Docker version 19.03.11, build dd360c7
 - docker-compose version 1.25.5, build unknown
 
 **Additional context**
 watchtower allocates 701M of virtual memory, a lot more than I expected, but only actually uses ~10M.

**Comment by kevinlul** [comment]
Unrelated but because of case-sensitivity, the code of conduct URL is dead 👆 . Actual URL: https://github.com/containrrr/watchtower/blob/master/code_of_conduct.md

**Comment by simonaronsson** [comment]
Thanks for the report!

This part of the actual compose file might lead to undefined behavior in general. What is it supposed to do?

```yank
    build: .
    image: "docker.pkg.github.com/alphakretin/emcee-tournament-bot/bot:${EMCEE_VERSION:-latest}"
```

Specifically, I'm interested in the combination of build and specifying an image.

**Comment by kevinlul** [comment]
With no images in the local cache, if you run `docker-compose up`, compose first tries to pull the specified image, and failing that, builds the image based on the specified context and tags it with the specified tag. If the tag already exists in the local cache, then compose will use that image, unless `--build` is specified.

`docker-compose pull` will try to pull the specified tag from the registry.
`docker-compose build` and `docker-compose up --build` will build the image based on the specified context and tag it with the specified tag.

More behaviour I believe can be tweaked by [pull policy](https://github.com/compose-spec/compose-spec/blob/master/spec.md#pull_policy) but I haven't used this feature.

**Comment by simonaronsson** [comment]
I spot two weird things here.

First, watchtower tries to clean up its own image. This shouldn't happen. Will need to check this.

Second, the actual segfault actually seems to be happening in shoutrrr. By the looks of it from `postURL` being nil. Would you mind verifying that the shoutrrr URL is actually set in your env vars? The changes included in v1.1.5 are https://github.com/containrrr/watchtower/compare/v1.1.4...v1.1.5, none of them changing anything related to notifications.

Shoutrrr has had commits since v1.1.4, but the go.sum locks it down to `78bb9685bc2f`, which is what the logs at https://app.circleci.com/pipelines/github/containrrr/watchtower/771/workflows/a440e902-da0b-42ce-bebb-fc476c8ca24b/jobs/2869 say as well. 🤔 @piksel, any other thoughts or ideas? 

As a side note, we should probably lock watchtower's shoutrrr usage down using `gopkg.in`, ie. https://gopkg.in/containrrr/shoutrrr.v0.

**Comment by kevinlul** [comment]
The shoutrrr URL is set in the environment variables, as that's what resulted in the Discord screenshot above and caused me to investigate!

**Comment by simonaronsson** [comment]
Ok, we've found two possible reasons that we're currently investigating. First one is that, for some reason, you never receive a ResponseStatus for the `Stopping /dKVlIoptpzKdNshheqXDgXROCDCKIYuL` post, which leads to shoutrrr trying to dereference a null value.

Second is, for some reason, watchtower is trying to do cleanups of its own images, which are still in use.

**Comment by simonaronsson** [comment]
Solved in `v1.1.6`
